### PR TITLE
fix: Updates the Dashboard Page URL

### PR DESCRIPTION
**Pull request by Golodhros**
### Summary of Changes
Updates the Dashboard Page URL to avoid depending on the query parameters
Flattened some components folder structure, removing /tests folders

### Tests
Updated tests

### CheckList
Make sure you have checked **all** steps below to ensure a timely review.
- [x] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
    - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [x] PR includes a summary of changes, including screenshots of any UI changes. 
- [x] PR adds unit tests, updates existing unit tests, __OR__ documents why no test additions or modifications are needed.
- [x] In case of new functionality, my PR adds documentation that describes how to use it.
    - All the public python functions and the classes in the PR contain docstrings that explain what it does
- [x] PR passes all tests documented in the [developer guide](https://github.com/lyft/amundsenfrontendlibrary/blob/master/docs/developer_guide.md#testing)

### CODE REVIEW

File: amundsen_application/static/js/components/TableDetail/index.spec.tsx [hunk 1]
```diff
@@ -10,27 +10,27 @@ import LoadingSpinner from '../common/LoadingSpinner';
...
-      statusCode: 200,
-      tableData: tableMetadata,
-      getTableData: jest.fn(),
-      ...routerProps,
-      ...propOverrides,
-    };
-    const wrapper = mount<TableDetail>(<TableDetail {...props} />);
-
-    return { props, wrapper };
+const setup = (propOverrides?: Partial<TableDetailProps>, location?: Partial<History.Location>) => {
```
Comment (ttannis) [inline amundsen_application/static/js/components/TableDetail/index.spec.tsx:13]: Any benefit to moving this outside, or is it perhaps just more readable this way?

File: amundsen_application/static/js/components/common/ResourceListItem/DashboardListItem/index.tsx [hunk 2]
```diff
@@ -23,7 +23,11 @@ export interface DashboardListItemProps {
 class DashboardListItem extends React.Component<DashboardListItemProps, {}> {
   getLink = () => {
     const { dashboard, logging } = this.props;
-    return `/dashboard?uri=${dashboard.uri}&index=${logging.index}&source=${logging.source}`;
+    const { cluster, product, group_url, uri } = dashboard;
+    const groupID = group_url.split('/').pop();
+    const dashboardName = uri.split('/').pop();
+
+    return `/dashboard/${cluster}/${product}/${groupID}/${dashboardName}?index=${logging.index}&source=${logging.source}`;
```
Comment (ttannis) [inline amundsen_application/static/js/components/common/ResourceListItem/DashboardListItem/index.tsx:30]: How did we come up with this specific path? If we use new concepts that the frontend has to parse and generate, then we end up introducing hardcoded url & uri processing logic that is specific to Mode. If community members are using some other tool, this parsing logic could pose issues for them unless all BI tools somehow format their values the same exact way. 

I would propose that we only use some combination of the exact values that come from the metadata service, and use `group_name` and `name` directly from our metadata.

File: amundsen_application/static/js/components/TableDetail/index.spec.tsx (hunk 1 above)
Comment (Golodhros) [reply amundsen_application/static/js/components/TableDetail/index.spec.tsx:13]: Yeah, I only see it as a readability matter. Is that OK?

Comment (ttannis) [reply amundsen_application/static/js/components/TableDetail/index.spec.tsx:13]: No problem

File: amundsen_application/static/js/components/common/ResourceListItem/DashboardListItem/index.tsx (hunk 2 above)
Comment (Golodhros) [reply amundsen_application/static/js/components/common/ResourceListItem/DashboardListItem/index.tsx:30]: This was a suggestion from @danwom, happy to adapt it.

I will follow your suggestion and move the URL creation into a helper so it is easy to change in the future.

**Comment by Golodhros** [comment]
> We also need to update the inline search result item that shows up in the search bar: https://github.com/lyft/amundsenfrontendlibrary/blob/master/amundsen_application/static/js/components/common/SearchBar/InlineSearchResults/index.tsx#L99
> 
> Beyond that, one concern about how we are forming the new url.

Thanks a lot for the heads up!
I will work on it!

File: amundsen_application/static/js/utils/navigationUtils.ts [hunk 3]
```diff
@@ -2,6 +2,7 @@ import * as qs from 'simple-query-string';
 import { createBrowserHistory } from 'history';
 
 import { ResourceType } from 'interfaces/Resources';
+import { MatchProps } from 'components/DashboardPage';
```
Comment (ttannis) [inline amundsen_application/static/js/utils/navigationUtils.ts:5]: nit: delete? don't see this type used here.

File: amundsen_application/static/js/components/common/SearchBar/InlineSearchResults/index.tsx [hunk 4]
```diff
@@ -5,6 +5,7 @@ import SearchItemList from './SearchItemList';
 import ResultItemList from './ResultItemList';
 
 import { getSourceDisplayName, getSourceIconClass, indexDashboardsEnabled, indexUsersEnabled } from 'config/config-utils';
+import { buildDashboardURL } from '../../../../utils/navigationUtils';
```
[the remark this answers is not in the document]
Comment (Golodhros) [reply amundsen_application/static/js/components/common/SearchBar/InlineSearchResults/index.tsx:8]: much better, changing!
Thanks!

File: amundsen_application/static/js/utils/navigationUtils.ts [hunk 5]
```diff
@@ -47,3 +48,16 @@ export const updateSearchUrl = (searchParams: SearchParams, replace: boolean = f
     BrowserHistory.push(newUrl);
   }
 };
+
+/**
+ * Creates the dashboard detail URL from the URI
+ * @param URI String  URI of the dashboard, it has this shape: uri = "<product>_dashboard://<cluster>.<groupID>/<dashboardID>"
+ * @return String     Dashboard Detail page URL
+ */
+export const buildDashboardURL = (URI: string) => {
```
[the remark this answers is not in the document]
Comment (Golodhros) [reply amundsen_application/static/js/utils/navigationUtils.ts:57]: Yep, writing one now.
Do you want me to add the product parameter already?

File: amundsen_application/static/js/utils/navigationUtils.ts (hunk 3 above)
Comment (Golodhros) [reply amundsen_application/static/js/utils/navigationUtils.ts:5]: Good call, it was used before.

Thanks

File: amundsen_application/static/js/utils/navigationUtils.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (ttannis) [reply amundsen_application/static/js/utils/navigationUtils.ts:56]: No need. We can add it when it's necessary, what's valuable is we made a decision that set us up to easily address the case without community members having to struggle too much to refactor anything.

**Pull request merged**